libs/vkd3d/device.c

```diff
@@ -5076,6 +5118,208 @@ struct d3d12_device *unsafe_impl_from_ID3D12Device(d3d12_device_iface *iface)
+    return d3d12_device_add_ref(device);
+}
+
+static ULONG STDMETHODCALLTYPE d3d12_device_vkd3d_ext_Release(ID3D12DeviceExt *iface)
+{
+    struct d3d12_device *device = d3d12_device_from_ID3D12DeviceExt(iface);
+    return d3d12_device_release(device);
+}
+
+static HRESULT STDMETHODCALLTYPE d3d12_device_vkd3d_ext_QueryInterface(ID3D12DeviceExt *iface,
+        REFIID iid, void **out)
+{
+    struct d3d12_device *device = d3d12_device_from_ID3D12DeviceExt(iface);
+    TRACE("iface %p, iid %s, out %p.\n", iface, debugstr_guid(iid), out);
+    return d3d12_device_QueryInterface(&device->ID3D12Device_iface, iid, out);
+}
+
+static HRESULT STDMETHODCALLTYPE d3d12_device_vkd3d_ext_GetVulkanHandles(ID3D12DeviceExt *iface, VkInstance *vk_instance, VkPhysicalDevice *vk_physical_device, VkDevice *vk_device)
+{
+    struct d3d12_device *device = d3d12_device_from_ID3D12DeviceExt(iface);
+    TRACE("iface %p, vk_instance %p, vk_physical_device %u, vk_device %p \n", iface, vk_instance, vk_physical_device, vk_device);
+    if (!vk_device || !vk_instance || !vk_physical_device)
+        return E_INVALIDARG;
+        
+    *vk_instance = device->vkd3d_instance->vk_instance;
+    *vk_physical_device = device->vk_physical_device;
+    *vk_device = device->vk_device;
+    return S_OK;
+}
+
+static BOOL STDMETHODCALLTYPE d3d12_device_vkd3d_ext_GetExtensionSupport(ID3D12DeviceExt *iface, D3D12_VK_EXTENSION extension)
+{
+    const struct d3d12_device *device = d3d12_device_from_ID3D12DeviceExt(iface);
+    bool ret_val = false;
+    
+    TRACE("iface %p, extension %u \n", iface, extension);
+    switch(extension)
+    {
+        case D3D12_VK_NVX_BINARY_IMPORT:
+            ret_val = device->vk_info.NVX_binary_import;
+            break;
+        case D3D12_VK_NVX_IMAGE_VIEW_HANDLE:
+            ret_val = device->vk_info.NVX_image_view_handle;
+            break;
+        default:
+            WARN("Invalid extension %x\n", extension);
+    }
+    
+    return ret_val;
+}
+
+static HRESULT STDMETHODCALLTYPE d3d12_device_vkd3d_ext_CreateCubinComputeShaderWithName(ID3D12DeviceExt *iface, const void *cubin_data,
+       UINT32 cubin_size, UINT32 block_x, UINT32 block_y, UINT32 block_z, const char *shader_name, D3D12_CUBIN_DATA_HANDLE **handle)
+{
+    VkCuFunctionCreateInfoNVX functionCreateInfo = { VK_STRUCTURE_TYPE_CU_FUNCTION_CREATE_INFO_NVX };
+    VkCuModuleCreateInfoNVX moduleCreateInfo = { VK_STRUCTURE_TYPE_CU_MODULE_CREATE_INFO_NVX };
+    const struct vkd3d_vk_device_procs *vk_procs;
+    struct d3d12_device *device;
+    VkDevice vk_device;
+    VkResult vr;
+    
+    TRACE("iface %p, cubin_data %p, cubin_size %u, shader_name %s \n", iface, cubin_data, cubin_size, shader_name);
+    if(!cubin_data || !cubin_size || !shader_name)
+        return E_INVALIDARG;    
+
+    device = d3d12_device_from_ID3D12DeviceExt(iface);
+    vk_device = device->vk_device;
+    *handle = vkd3d_calloc(1, sizeof(D3D12_CUBIN_DATA_HANDLE));
```

Comment:
Would be cleaner to declare a local `*handle` and parameter is `**out_handle` with a `*out_handle = handle;` at the end.